release.yml

```diff
@@ -3,88 +3,177 @@ name: $(Date:yyyyMMdd)$(Rev:.r)
-    pathToPublish: '$(Build.BinariesDirectory)/tool' 
-    artifactName: 'FhirConverterBuild'
-
-- task: PublishBuildArtifacts@1
-  displayName: 'publish nuget artifacts'
-  inputs:
-    pathToPublish: '$(Build.ArtifactStagingDirectory)/nupkgs'
-    artifactName: 'nuget'
-    publishLocation: 'container'
\ No newline at end of file
+    - task: PublishBuildArtifacts@1
+      displayName: 'publish nuget artifacts'
+      inputs:
+        pathToPublish: '$(Build.BinariesDirectory)/nupkgs'
+        artifactName: 'nuget'
+        publishLocation: 'container'     
+    
+    - task: PublishBuildArtifacts@1
+      displayName: 'publish artifacts'
+      inputs:
+        pathToPublish: '$(Build.ArtifactStagingDirectory)' 
+        artifactName: 'FhirConverterBuild'
+
+- stage: FunctionalTests
+  jobs:
+  - job: Windows_Functional_Test
+    pool:
+        vmImage: 'windows-latest'
+    continueOnError: true
+    steps:
+    - checkout: none #skip checking out the default repository resource
+    - task: DownloadBuildArtifacts@0
+      displayName: 'Download Artifacts'
+      inputs:
+        artifactName: FhirConverterBuild
+        downloadPath: $(System.DefaultWorkingDirectory)
+    - script: |
+        docker run --rm -d -p 5000:5000 --name registry stefanscherer/registry-windows:2.6.2
+      displayName: start registry
+
+    # - script: |
+    #     docker pull mcr.microsoft.com/windows/servercore:ltsc2019
+    #     docker tag mcr.microsoft.com/windows/servercore:ltsc2019 localhost:5000/windows
+    #     docker push localhost:5000/windows
+    #   displayName: test push
+
+    - script: |
+        dotnet test **/*FunctionalTests.dll
+      displayName: Functional Tests
+    - task: Docker@2
+      displayName: stop registry
+      condition: succeededOrFailed()
+      inputs:
+        command: stop
+        container: registry
+
+  - job: Linux_Functional_Test
+    pool:
+      vmImage: 'ubuntu-latest'
+    continueOnError: true
+    steps:
+    - checkout: none #skip checking out the default repository resource
+    - task: DownloadBuildArtifacts@0
+      displayName: 'Download Artifacts'
+      inputs:
+        artifactName: FhirConverterBuild
+        downloadPath: $(System.DefaultWorkingDirectory)
+    - script: |
+        docker run --rm -d -p 5000:5000 --name registry registry:2
+      displayName: start registry
+    - script: |
+        dotnet test **/*FunctionalTests.dll
+      displayName: Functional Tests
+    - task: Docker@2
+      displayName: stop registry
+      condition: succeededOrFailed()
+      inputs:
+        command: stop
+        container: registry
+  
+- stage: Release
+  jobs:
+  - job: publish
+    pool:
+        vmImage: 'windows-latest'
+    steps:
+    - checkout: none #skip checking out the default repository resource
+    - task: DownloadBuildArtifacts@0
+      displayName: 'Download Artifacts'
+      inputs:
+        artifactName: FhirConverterBuild
+        downloadPath: $(System.DefaultWorkingDirectory)
```

Comment:
Are we going to download the artifacts into the vm?

Reply:
yes